exporter/datadogexporter/factory.go

```diff
@@ -156,7 +156,8 @@ func createMetricsExporter(
 		pushMetricsFn,
 		// explicitly disable since we rely on http.Client timeout logic.
 		exporterhelper.WithTimeout(exporterhelper.TimeoutSettings{Timeout: 0 * time.Second}),
-		exporterhelper.WithRetry(cfg.RetrySettings),
```

Comment:
This looks like a breaking change. Could you add this to the changelog? Make sure also to add this to the readme. Is there a way to determine whether the retry settings were customized by users, and warn them?

Reply:
Retry settings will still be used here https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/3e21ff75d070d1677c57965d3cfcfc1b4dd5ae1f/exporter/datadogexporter/internal/utils/retrier.go#L53-L61, what this PR changes is that we now retry on each network call we do, instead of retrying the whole `ConsumeMetrics` function (so, we disable the `exporterhelper` retry mechanism). I believe this change should be transparent to users and it will probably align closer with what one would expect from the retry settings.

Reply:
Got it! Makes sense then.